webpack/worker/worker_adapter.js

```diff
@@ -28,118 +81,103 @@ function main() {
   }
 }
 
-/**
- * Returns arguments that is preceded by -c
- * @param {string[]} args
- * @returns {string[]}
- */
-function findWebpackConfigs(args) {
-  const configs = []
-  for (let i = 0; i < args.length; i++) {
-    if (i > 0 && args[i - 1] == '-c') {
-      configs.push(args[i])
-    }
-  }
-  return configs
-}
-
-function runAsPersistentWorker() {
-  const webpackCliPath = require.resolve('webpack-cli/bin/cli.js')
-  /** @type {string} */
-  let key
-  /** @type {cp.ChildProcess} */
-  let proc
-
-  /** @type {Map<string, string>} */
-  let configDigestMap = new Map()
+async function runAsPersistentWorker() {
+  const cli = new WorkerAwareCLI()
+  /** @type {import("@bazel/worker").Inputs} */
```

Comment:
optional: In order to get typechecking in this file we could make it into an actual typescript file and have a rule check in the transpiled result from `tsc`

Up to you, but I thought I'd mention it since the file is getting more complex

Reply:
I agree. However, tradeoffs make me wonder if it is actually worth it. I don't give up on the simplicity of this repository.

Maybe we should keep this as `.js` and introduce a test target (skylib/build_test) that only typechecks this file with `checkJs`?

if you believe that could serve the same purpose, we can always do that.